app/2.0/docs/devguide/encapsulated-style.md

```diff
@@ -0,0 +1,639 @@
+
+```html
+<dom-module id="x-foo">
+    <template>
+      <style>
+        ::slotted(*) {
+          font-family: sans-serif;
+          color:green;
+        }
+      </style>
+        <h1>
+            <div><slot name='heading1'></slot></div>
+        </h1>
+        <p>
+            <slot name='para'></slot>
+        </p>
+    </template>
+</dom-module>
+<x-foo>
+    <div slot="heading1">Heading 1. I'm green.</div>
+    <div slot="para">Paragraph text. I'm green too.</div>
+</x-foo>   
+```
+
+[See it on Plunkr](http://plnkr.co/edit/suVhRDAB5Rq7Q7T6zbMI?p=info)
+
+You can select by element type:
+
+```html
+<dom-module id="x-foo">
+	<template>
+		<style>
+		::slotted(h1) {
+			font-family: sans-serif;
+			color:green;
+		}
+		::slotted(p) { 
+			font-family: sans-serif;
+          color:blue;
+      }
+      </style>    
+            <slot name='heading1'></slot>
+            <slot name='para'></slot>
+    </template>
+</dom-module>
+<x-foo>
+    <h1 slot="heading1">Heading 1. I'm green.</h1>
+    <p slot="para">Paragraph text. I'm blue.</p>
+</x-foo>
+```
+
+You can select by class:
+
+[See it on Plunker](http://plnkr.co/edit/Ep8AVOHgiwQjtv8x5kwd?p=preview)
+
+```
+<dom-module id="x-foo">
+    <template>
+      <style>
+        ::slotted(.green) {
+          color:green;
+        }
+      </style>
+        <p>
+            <slot name='para1'></slot>
+        </p>
+        <p>
+            <slot name='para2'></slot>
+        </p>
+    </template>
+</dom-module>
+
+<x-foo>
+    <div slot="para1" class="green">I'm green!</div>
+    <div slot="para1">I'm not green.</div>
+    <div slot="para2" class="green">I'm green too.</div>
+    <div slot="para2">I'm not green.</div>
+</x-foo> 
+```
+
+And you can select by slot name:
+
+[See it on Plunker](http://plnkr.co/edit/FnFLnownBJc0Ur8vC6dd?p=preview)
+
+```
+<dom-module id="x-foo">
+    <template>
+      <style>
+        slot[name='para1']::slotted(*) {
+          color:green;
+        }
+      </style>
+        <p>
+            <slot name='para1'></slot>
+        </p>
+        <p>
+            <slot name='para2'></slot>
+        </p>
+    </template>
+</dom-module>
+
+<x-foo>
+    <div slot="para1">I'm green!</div>
+    <div slot="para2">I'm not green.</div>
+</x-foo> 
+```
+
+For more examples, see [Rob Dodson's Shadow DOM CSS Cheat Sheet](http://robdodson.me/shadow-dom-css-cheat-sheet/).
+
+## Sharing styles
+
+### Using style modules {#style-modules}
+
+The preferred way to share styles is with *style modules*. You can package up styles in a style module, and share them between elements.
+
+To create a style module, wrap your style block in `<dom-module>` and `<template>` elements, like this:
+
+```html
+<dom-module id="my-style-module">
+	<template>
+		<style>
+			<!-- Styles to share go here! -->
+		</style>
+	</template>
+</dom-module>
+```
+
+When you create the element that will use the styles, include the style module in the opening tag of the style block:
```

Comment:
An important missing bit here is that you almost invariably define the style module in a different file than you use it in. So you should show the style module with a filename (`my-style-module.html`) and show the custom element file (`new-element.html`) importing the style module at the top using a `<link rel="import" href="my-style-module.html">` (or whatever path you like).